### PR TITLE
Fix modules to support new Apigee X environment types

### PR DESCRIPTION
**Pull request by Teodelas**
---
**Checklist**

I applicable, I acknowledge that I have:
- [X] Read the [contributing guide](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md)
- [ X] Ran `terraform fmt` on all modified files
- [ ] Regenerated the relevant README.md files using [`tools/tfdoc.py`](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md#fabric-tools)
- [X] Made sure all relevant tests pass

### CODE REVIEW

File: modules/apigee/variables.tf [hunk 1]
```diff
@@ -50,6 +50,7 @@ variable "environments" {
     description     = optional(string, "Terraform-managed")
     deployment_type = optional(string)
     api_proxy_type  = optional(string)
+    type = optional(string)
```
Comment (wiktorn) [inline modules/apigee/variables.tf:53]: Can you sort members alphabetically? We usually do that, but for some reason, it slip under the radar here.

**Review by wiktorn — APPROVED** [review]
Thank you for your contribution.

You will need to run `terraform fmt` on modified files to make the lint checks pass. Also, you need to update the README's, by:
```
$ python3 -m venv venv
$ . venv/bin/activate
$ pip installl -r tools/requirements.txt
$ python tools/tfdoc.py modules/apigee
```

The last thing is, that this requires provider in version 5.4.0, and we are still testing against 5.0.0, hence the failures. I'll create a separate PR to bump the version.

File: modules/apigee/variables.tf (hunk 1 above)
Comment (Teodelas) [reply modules/apigee/variables.tf:53]: Sure!  Thanks, first time contributing to this repo so bear with me. :-)

**Review by ludoo — CHANGES_REQUESTED** [review]
please don't commit your virtualenv

**Comment by Teodelas** [comment]
> please don't commit your virtualenv

Sorry, didn't realize that. removed the directory.

**Review by ludoo — APPROVED** [review]

**Pull request merged**